resilience4j-spring/src/main/java/io/github/resilience4j/circuitbreaker/configure/CircuitBreakerAspect.java

```diff
@@ -97,7 +98,10 @@ public Object circuitBreakerAroundAdvice(ProceedingJoinPoint proceedingJoinPoint
 			return proceed(proceedingJoinPoint, methodName, circuitBreaker, returnType);
 		}
 
-		FallbackMethod fallbackMethod = new FallbackMethod(circuitBreakerAnnotation.fallbackMethod(), method, proceedingJoinPoint.getArgs(), proceedingJoinPoint.getTarget());
+		FallbackMethod fallbackMethod = FallbackMethod.builder().recoveryMethodName(circuitBreakerAnnotation.fallbackMethod())
```

Comment:
Is anything optional? I think every parameter (recoveryMethodName, originalMethod, targetObject) is mandatory or?
The builder pattern should be used if some parameters are optional.

Reply:
The building of Fallback method is not straight forward and i did not like that we have logic in the constructor , but u are right with the optional params with the builder pattern , i replaced it with simple static factory method now